src/main/java/io/seqera/tower/cli/commands/AbstractApiCmd.java

```diff
@@ -174,6 +174,46 @@ protected ComputeEnv computeEnvByName(Long workspaceId, String name) throws ApiE
         throw new TowerException(String.format("Compute environment '%s' is not available", name));
     }
 
+    protected ComputeEnv computeEnvById(Long workspaceId, String id) throws ApiException {
+        if (availableComputeEnvsNameToId == null) {
+            loadAvailableComputeEnvs(workspaceId);
+        }
+
+        if (availableComputeEnvsNameToId.containsValue(id)) {
+            String name = availableComputeEnvsNameToId.entrySet().stream()
+                    .filter(it -> Objects.equals(it.getValue(), id))
+                    .map(Map.Entry::getKey)
+                    .findFirst()
+                    .orElse(null);
+
+            return api().describeComputeEnv(availableComputeEnvsNameToId.get(name), workspaceId).getComputeEnv();
+        }
+
+        throw new TowerException(String.format("Compute environment '%s' is not available", id));
+    }
+
+    protected ComputeEnv computeEnvByRef(Long workspaceId, String ref) throws ApiException {
+        if (availableComputeEnvsNameToId == null) {
```

Comment:
Maybe you can use here the `computeEnvById` method and catch the exception to try the resolution by name.